--- a/frameworks/native/libs/ui/PixelFormat.cpp
+++ b/frameworks/native/libs/ui/PixelFormat.cpp
@@ -162,6 +162,13 @@
     return BAD_VALUE;
 }
 
+
+extern "C" void _ZN7android18getPixelFormatInfoEiPNS_15PixelFormatInfoE(void* obj, size_t size, const char* name, uint32_t flags);
+extern "C" void _ZN7android18getPixelFormatInfoEiPNS_15PixelFormatInfo(void* obj, size_t size, const char* name) {
+    _ZN7android18getPixelFormatInfoEiPNS_15PixelFormatInfoE(obj, size, name, 0);
+}
+
+
 // ----------------------------------------------------------------------------
 }; // namespace android
 // ----------------------------------------------------------------------------
